### ISSUE TITLE
[Bug]: Playwright UI timeline data get wiped when test fails

### ISSUE THREAD

**Issue by Trystan-SA** [microsoft/playwright#30753]
### Version

1.44.0 & 1.43.0

### Steps to reproduce

Run a simple failing test.

### Expected behavior

I expect to see the timeline data even if the test fail.

### Actual behavior

The timeline data get wiped. I can't inspect it when a test fail (but work well when the test succeed).

### Additional context

I'm using playwright UI to troubleshot my E2E tests. When I'm running a test, it successfully load the timeline (middle upper-part, where you can see the browser doing its manipulations when running the tests).

![mLhEgzID](https://github.com/microsoft/playwright/assets/43876044/f3a6bc5c-fd97-4329-afd0-bd3bfb3b033f)

But when my test fail, the timeline and actions panels suddenly get empty, I can't troubleshoot my tests because of that...

![pBtP0DLf](https://github.com/microsoft/playwright/assets/43876044/0e73b7bf-3b1f-4619-bf9e-e765e87b6e64)

I previously used playwright 1.35.1, but needed to upgrade to fix some issues. I'm now using 1.44.0, same issue with 1.43.0.

I'm running npx playwright test --ui inside a WSL container and shell.

Thanks for your help !

### Environment

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz
    Memory: 13.39 GB / 15.51 GB
    Container: Yes
  Binaries:
    Node: 21.7.2 - ~/.nvm/versions/node/v21.7.2/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v21.7.2/bin/npm
  IDEs:
    VSCode: 1.89.1 - /home/trystan/.vscode-server/bin/dc96b837cf6bb4af9cd736aa3af08cf8279f7685/bin/remote-cli/code
  Languages:
    Bash: 5.1.16 - /usr/bin/bash
  npmPackages:
    @playwright/test: 1.43.0 => 1.43.0
```

**Comment by mxschmitt** [comment]
`1.43.0` had a related bug, we fixed it in `1.43.1`. `1.44.0` should have the fix as well.

Could you verify that you have the right version via `npx playwright --version`?

Would it be possible to share a minimal reproducible? I tried the following which worked for me:

```ts
import { test  } from "@playwright/test";

test("test", async ({ page }) => {
  await page.goto("https://www.google.com");
  throw new Error("This is a test error");
});
```

**Comment by yury-s** [comment]
Closing per the response above, feel free to open a new issue if it doesn't work.
